rs/nns/governance/src/governance.rs

```diff
@@ -1105,6 +1109,19 @@ impl ProposalData {
         now_seconds < self.get_deadline_timestamp_seconds(voting_period_seconds)
     }
 
+    /// Returns if voting is closed along with whether or not any outstanding votes still
+    /// need to be cast (when processing following spans multiple messages).
+    pub fn voting_is_finished(&self, now_seconds: u64, voting_period_seconds: u64) -> bool {
```

Comment:
Note for reviewer: This is a new concept for proposals, please look at this, along with new proposal status usage.